### PR TITLE
apps sc: Grafana upgrade v10.3.1

### LINKED ISSUE #1907: [3] Upgrade Grafana
**What Chart should be upgraded?**:

At least app [v10.2.2](https://github.com/grafana/grafana/tree/v10.2.2) and [upstream/grafana](https://github.com/grafana/helm-charts/blob/grafana-7.0.11/charts/grafana/Chart.yaml), preferably the latest stable version (if no big breaking changes for us)

**Why should this Chart be upgraded?**:

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**:

- I checked the migration of the new Chart:
  - [ ] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by robinelastisys**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [ ] personal data beyond what is necessary for interacting with this pull request
> - [ ] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

- Fixes #1907 

Resource increase is needed as the sidecar container needs extra resources to show all dashboards, or else it will go OOM.

No increase:
Screenshot from 2024-02-12 08-52-16

With increase:
Screenshot from 2024-02-12 08-55-15

#### Additional information to reviewers
https://grafana.com/docs/grafana/latest/whatsnew/whats-new-in-v10-0/

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [x] Any changed pod is covered by Pod Security Admission
  - [x] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [x] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [x] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: config/config/sc-config.yaml [hunk 1]
```diff
@@ -129,6 +129,14 @@ grafana:
     #     timeout: 600
   user:
     enabled: true
+    sidecar:
```
Comment (crssnd) [inline config/config/sc-config.yaml:132]: this is not needed, as the values should be templated from the grafana.ops key

File: config/config/sc-config.yaml [hunk 2]
```diff
@@ -114,11 +114,11 @@ grafana:
     sidecar:
       resources:
         requests:
-          cpu: 10m
-          memory: 80Mi
+          cpu: 80m
+          memory: 640Mi
```
Comment (crssnd) [inline config/config/sc-config.yaml:118]: this is a huge increase in memory requests. is this needed?

**Comment by crssnd** [comment]
please fill in the check list at the end of the PR so we know what was tested

**Comment by crssnd** [comment]
I do not see any app developer or system administrator notes. Are there no upstream changes worth mentioning?

File: config/config/sc-config.yaml (hunk 1 above)
Comment (robinelastisys) [reply config/config/sc-config.yaml:139]: It looks to me as if "ops" and "user" have different values if im not mistaken?:

```
resources: {{- toYaml .Values.grafana.user.sidecar.resources | nindent 6 }}

resources: {{- toYaml .Values.grafana.ops.sidecar.resources | nindent 6 }}
```

File: config/config/sc-config.yaml (hunk 2 above)
Comment (robinelastisys) [reply config/config/sc-config.yaml]: Yep, for sure was a bit questionable at first. Have tried to narrow it down now.

File: config/config/sc-config.yaml (hunk 1 above)
Comment (crssnd) [reply config/config/sc-config.yaml:139]: the values are templated from the ops section (see `<<: *grafanadefaults` below)

Comment (robinelastisys) [reply config/config/sc-config.yaml]: Thanks, took your directions and it worked out fine!

**Comment by robinelastisys** [comment]
> please fill in the check list at the end of the PR so we know what was tested

Done!

> I do not see any app developer or system administrator notes. Are there no upstream changes worth mentioning?

Not really anything special. Feels as if it is more of a view upgrade from our perspective. You can check if you think something is important: https://grafana.com/docs/grafana/latest/breaking-changes/breaking-changes-v10-0/

**Review by crssnd — APPROVED** [review]

File: config/config/sc-config.yaml [hunk 3]
```diff
@@ -115,10 +115,10 @@ grafana:
       resources:
         requests:
           cpu: 10m
-          memory: 80Mi
+          memory: 150Mi
         limits:
-          cpu: 50m
-          memory: 100Mi
+          cpu: 250m
+          memory: 200Mi
```
Comment (Xartos) [inline config/config/sc-config.yaml:121]: Is there a reason for this change as well? Like will it not start if this is set to the previous value or is it just unusably slow?

Comment (robinelastisys) [reply config/config/sc-config.yaml:121]: Yeah, it gets super slow.. Then we had a customer situation that made us aware that we may need a limit at 250m.

**Review by Xartos — APPROVED** [review]

**Pull request merged**